server/dataset/cxg_dataset.py

```diff
@@ -403,21 +402,52 @@ def get_uns(self, metadata_key):
 
     def get_atac_coverage(self, gene_name, genome_version, cell_type):
         """
-        Extracts ATAC coverage data - currently random mock data
+        Extracts ATAC coverage data for a gene and cell type from TileDB.
         """
 
-        # get gene info
-        target_chromosome, sorted_genes, range_start, range_end = self.get_atac_gene_info(gene_name, genome_version)
+        def get_gene_start_end(gene_name, genes):
+            for gene in genes:
+                if gene["geneName"] == gene_name:
+                    return gene["geneStart"], gene["geneEnd"]
+            return None
 
-        # get coverage data (currently random mock data)
+        coverage = self.open_array("coverage")
         bin_size = 100
-        coverage_plot = []
-        for start in range(range_start, range_end, bin_size):
-            end = min(start + bin_size, range_end)
-            coverage = random.randint(0, 100)  # mock read count
-            coverage_plot.append([coverage, start, end])
+        target_chromosome, sorted_genes = self.get_atac_gene_info(gene_name, genome_version)
+
+        gene_start, gene_end = get_gene_start_end(gene_name, sorted_genes)
+
+        range_buffer = 10_000  # 1 Mbp buffer around the gene
```

Comment:
super nit: this is just 10K basepairs around the gene not 1 million basepairs (`1 Mbps`), right lol?